tests/Fixtures/project/src/Aspect/InconsistentlyWeavingAspect.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Go\Tests\TestProject\Aspect;
+
+use Go\Aop\Aspect;
+use Go\Aop\Intercept\MethodInvocation;
+use Go\Tests\TestProject\Application\InconsistentlyWeavedClass;
+use Go\Lang\Annotation as Pointcut;
+
+class InconsistentlyWeavingAspect implements Aspect
+{
+    public function __construct(InconsistentlyWeavedClass $problematicClass)
```

Comment:
Ah, yes )) Injection of intercepted class, please leave a note about this edge-case in the doc block and link to the issue on github why this happen.